rh-summit-2018/module-03.adoc

```diff
@@ -1,3 +1,302 @@
-== Running hiking demo with Kafka Connect and Debezium
+== Change Data Capture with Kafka Connect and Debezium
 
-TBD
\ No newline at end of file
+This part of the lab introduces you to change data capture (CDC) using http://debezium.io/[Debezium].
+Debezium allows you to capture data changes from MySQL, Postgres and MongoDB and stream those events into Apache Kafka.
+It is based on Kafka Connect.
+
+In the following you'll learn the following things:
+
+* Setting up Kafka Connect with the Debezium CDC connectors
+* Setting up the Debezium MySQL connector, ingesting changes from an existing example CRUD application
+* Streaming Data Changes to a Postgres Sink Database using the Confluent JDBC sink connector
```

Comment:
Are we going to use Confluent stuff at the RH summit ? :-(

Reply:
It's the commonly used JDBC sink connector for Kafka Connect. I think it's fair to show/use it.

Reply:
Can we use a Camel application instead though?

Reply:
We could; but don't you think we should show a pure Kafka Connect (i.e. no code) pipeline in some way? It's useful for many use cases and IMO the more common usage, compared to more custom receivers. Maybe using another sink connector if you're truly concerned about the JDBC sink one?

Also my experience with Camel dates back quite some time, so some help with that would be good if we really want to go for that. IIRC, @jpechane has some experience here.

Finally, together with the Swarm sink application, I'm wondering whether that's not all getting a bit too much for two hours?

Reply:
I agree with @gunnarmorling about not adding more stuff as Camel to the demo but I'm still a bit skeptical about using the JDBC one. I have never used Debezium (sorry for that !) so a stupid question is coming : doesn't Debezium provide a PostgreSQL connector as I see from the project web page ?

Reply:
@ppatierno Debezium provides only source connectors, not sink (write) ones. That's why we use JDBC sink connector.

@mbogoevici I am a fan of Camel but I'd keep this demo relate only to Kafka ecosystem - so a simple Kafka pipeline. IMHO Camel is suitable when you need further functionality.

Reply:
@jpechane I understand how this can be a simple solution, and I'm even fine with the idea keeping it this way, at least for now. 

As a side comment though, I believe that the premise of 'Camel is suitable when you need further functionality' is flawed (or at least too restrictive): streaming change events to a data store is an enterprise integration scenario, and Camel has a rich (and production-tested) ecosystem of integration endpoints. Not related to this lab, but something to consider once, we will, for example, support other messaging systems than Kafka.

Reply:
> but this being the Summit I still feel that we should try to use components that we expect customers to use in a real-world setting

I think using the JDBC sink is a very, very real-world setting.

> I also think we want to encourage them to explore Camel

Yeah, we can do that.

So how about that: we offer the three scenarios on the sink side for the lab:

* consuming messages with the JDBC sink connector
* consuming them with the WF Swarm app + WebSockets
* consuming them with Camel

Attendees could choose which of them (or all) they'd like to explore. If we wanted to do that, it'd be great though if someone else could help with preparing the Camel part, as my experience with it is quite limited (and probably outdated by now).

WDYT?

Reply:
> streaming change events to a data store is an enterprise integration scenario, and Camel has a rich (and production-tested) ecosystem of integration endpoints.

True, but it also adds another moving piece to the puzzle. As a user, if I wanted to simply replicate data into another DB and I could do this sufficiently well with KC and its available connectors alone, I'd prefer that instead of adding one more piece of technology which I need to understand and operate.

> Not related to this lab, but something to consider once, we will, for example, support other messaging systems than Kafka.

Absolutely. Note I'm not against integrating with Camel at all, if it's the right fit for a given use case. And I know it has been on @jpechane's heart for a long time to explore the integration :) It just appeared as "too much" for the lab, but happy to offer it as an option as suggested above.

Reply:
I can volunteer for the Camel part but if I were you I would not introduce it now as it is another burden to solve during the lab if anything is not going well.

Reply:
I don't want to be so pessimistic but I don't think that we'll have enough time for showing different types of sinks. I think that just one is enough, let's discuss about it in the lab sync up today. For sure adding the Camel part is really useful for improving this lab that can be used for other demos in the future.